Gem/Code/Source/SampleComponentManager.cpp

```diff
@@ -206,11 +228,80 @@ namespace AtomSampleViewer
         dependent.push_back(AZ_CRC("AzFrameworkConfigurationSystemComponentService", 0xcc49c96e)); // Ensures a scene is created for the GameEntityContext
     }
 
+    const AZStd::vector<SampleEntry>& SampleComponentManager::GetSamples()
+    {
+        static AZStd::vector<SampleEntry> sampleEntries{
```

Comment:
I don't think you are allowed to make a static vector because it uses allocators that are lifetime managed. Since this function is only called once, I suggest you just construct and return a copy.

Reply:
In this case it works because this static variable is scoped within a function and this is called after the module itself initializes. Since this vector is used twice (once to query component descriptors on module entry, once to initialize the sample component needed to build the menus), I'd rather not have to build it twice if not needed. Thoughts?

Reply:
Hmm, how about the order during program termination? If the allocator is destroyed before static variable is destroyed?

Reply:
I believe what happens at program exit is that entire virtual memory pages are returned back to the OS. That said I don't mind just changing this to make the storage completely static.

Reply:
This is done now, no allocations on startup